exoctk/exoctk_app/app_exoctk.py

```diff
@@ -359,6 +376,30 @@ def pa_contam():
     return render_template('pa_contam.html')
 
 
+# Long-running Celery task
+@celery.task
+def run_contam_visibility_task(params):
```

Comment:
Looks like you haven't implemented the task to tun the contam tool quite yet? Did you mean to call the contam tool in place time with the params supplied by the user in the form?

Reply:
Oh yeah, haha. Not sure how I avoided the timeout in my test though! Line 528 is the task that needs to be put into `run_contam_visibility_task`.  Let me update it and test it.